qiskit_nature/hdf5.py

```diff
@@ -0,0 +1,160 @@
+        attributes of an HDF5 group. Furthermore, a `__version__` should be stored in order to allow
+        version handling at runtime.
+
+        Args:
+            parent: the parent HDF5 group.
+        """
+        ...
+
+    @classmethod
+    def from_hdf5(cls, h5py_group: h5py.Group) -> Any:
+        """Constructs a new instance from the data stored in the provided HDF5 group.
+
+        Args:
+            h5py_group: the HDF5 group from which to load the data.
+
+        Returns:
+            A new instance of this class.
+        """
+        ...
+
+
+def save_to_hdf5(obj: HDF5Storable, filename: str, force: bool = False) -> None:
+    """A utility to method to store an object to an HDF5 file.
+
+    .. code-block:: python
+
+        property = driver.run()
+        save_to_hdf5(property, "my_driver_result.hdf5")
+
+    Args:
+        obj: the `HDF5Storable` object to store in the file.
+        filename: the path to the HDF5 file.
+        force: whether to forcefully overwrite an existing file.
+
+    Raises:
+        QiskitNatureError: if the file at the given path already exists and forcefully overwriting
+            is not enabled.
+    """
+    if not isinstance(obj, HDF5Storable):
+        LOGGER.error("%s is not an instance of %s", obj, HDF5Storable)
+        return
+
+    if Path(filename).exists() and not force:
+        raise QiskitNatureError(
+            f"The file at {filename} already exists! Specify `force=True` if you want to overwrite it!"
+        )
+
+    with h5py.File(filename, "w") as file:
+        obj.to_hdf5(file)
+
+
+def load_from_hdf5(filename: str) -> Generator[Any, None, None]:
+    """Loads Qiskit Nature objects from an HDF5 file.
+
+    .. code-block:: python
+
+        my_driver_result = load_from_hdf5("my_driver_result.hdf5")
+
+    Args:
+        filename: the path to the HDF5 file.
+
+    Yields:
+        The objects constructed from the HDF5 groups encountered in the h5py_group.
+    """
+    with h5py.File(filename, "r") as file:
+        yield from _import_and_build_from_hdf5(file)
+
+
+def _import_and_build_from_hdf5(h5py_group: h5py.Group) -> Generator[Any, None, None]:
+    """Imports and builds a Qiskit Nature object from an HDF5 group.
+
+    Qiskit Nature uses the convention of storing the `__module__` and `__class__` information as
+    attributes of an HDF5 group. From these, this method will import the required class at runtime
+    and use its `form_hdf5` method to construct an instance of the encountered class.
+
+    Args:
+        h5py_group: the HDF5 group from which to import and build Qiskit Nature objects.
+
+    Yields:
+        The objects constructed from the HDF5 groups encountered in the h5py_group.
+    """
+    for group in h5py_group.values():
+        module_path = group.attrs.get("__module__", "")
+        if not module_path:
+            continue
+
+        class_name = group.attrs.get("__class__", "")
+
```

Comment:
I am currently logging+skipping am I not? I think in this specific case this is the better choice in order to attempt to recover as much from the file as possible.

Reply:
Yes you are logging/skipping. Sorry if my comment was less than clear, but I meant maybe instead we should be raising errors by default? So that bits are not skipped over by default, which may potentially lead to problems. If a user wants to try to try to get what they can from a file, that cannot be fully processed, maybe that should be enabled by them explicitly? Certainly there can be cases where parts are optional to the problem - but what if the data that we cannot process is really needed, does this then lead to errors later that are harder to diagnose. I am guessing that normally for the stack this is not going to occur - perhaps if/when other custom properties come about where we do not understand all the data this might happen.

Reply:
Ah okay. Yes that can be done. We can raise errors by default but provide a keyword-argument to the hdf5 methods along the lines of `raising: bool = True` which a user can disable to drop back to logging error instead of raising them.
Maybe you come up with a better name than `raising` though..

Reply:
I am not sure - something more like `skip_unknown_data=False`, so it raises an error by default, though maybe `unknown` here may not be the best choice. Perhaps `skip_unreadable_data=False` (unprocessable seems too long). Either way unknown or unreadable was meant to mean either we do not have the class, or refuse/cannot use it, to read the data.

I imagine at the end of the day this would be a parameter into the new hdf5 driver that on reading a file, if it cannot access all the data it raises and error, or optionally continues with what it can read noting in the log those sections it had to ignore.